### PR TITLE
Moderation warning

### PR DESCRIPTION
**Pull request by Raemon**
This PR lets a forumMagnum instance set particular comments as the text for a "comment moderation warning" and "post moderation warning".

If the setting is _not_ set, it looks like this on posts:

image

and this on comments:

image

Once set, you can customize the text however you want. Examples from the current draft on LessWrong are:

image

image

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204377807093813) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/posts/PostsNewForm.tsx [hunk 1]
```diff
@@ -230,12 +231,7 @@ const PostsNewForm = ({classes}: {
     <div className={classes.postForm}>
       <RecaptchaWarning currentUser={currentUser}>
         <Components.PostsAcceptTos currentUser={currentUser} />
-        {postWillBeHidden && <ContentStyles contentType="comment" className={classes.modNote}>
```
Comment (darkruby501) [inline packages/lesswrong/components/posts/PostsNewForm.tsx:233]: Was this not forumGated  before?

**Review by darkruby501 — APPROVED** [review]

File: packages/lesswrong/components/posts/PostsNewForm.tsx (hunk 1 above)
Comment (Raemon) [reply packages/lesswrong/components/posts/PostsNewForm.tsx:233]: it was – the forumgating lived in the postWillBeHidden field.  (But, I think probably for this PR it'd probably be net positive for EA Forum to remove the forumGating, and showing the default "moderators will need to approve this" message. @jpaddison3 interested in your take)

**Comment by jpaddison3** [comment]
Have not reviewed yet, but in principle open to this on the Forum. We have something like this already, no? But this seems good still.

**Comment by Raemon** [comment]
@jpaddison3 I am merging this for now. From your perspective it should be a relatively minimal styling change that I don't think you'll consider consequential, but, if you are like "nope I hate that styling change" happy to make a followup PR forumGating it.

**Pull request merged**